txnsync/emulatorNode_test.go

```diff
@@ -75,6 +77,226 @@ type emulatedNode struct {
 	nodeRunning                         chan struct{} // channel is closed when node is running.
 }
 
+type proposalData struct {
+	_struct struct{} `codec:",omitempty,omitemptyarray"`
+
+	ProposalBytes []byte              `codec:"b,allocbound=maxNumProposalBytes"`
+	TxGroupIds    []transactions.Txid `codec:"h,allocbound=maxNumTxGroupHashesBytes"` // TODO: make this []byte
+}
+
+const maxNumProposalBytes = 30000       // sizeof(block header)
+const maxNumTxGroupHashesBytes = 320000 // 10K * 32
+
+// MarshalMsg implements msgp.Marshaler
+func (z *proposalData) MarshalMsg(b []byte) (o []byte) {
```

Comment:
Is there a better way of doing this other than copying the data structure and its encoding and decoding methods

Reply:
yes - create a child package inside txnsync called "testing". Create the file there and have it msgp'ed. then, import this package only from the test.